python/ccf/ledger.py

```diff
@@ -264,18 +289,13 @@ def __init__(self, directory: str):
                     LOG.warning(f"The file {chunk} has not been committed")
                 self._filenames.append(os.path.join(directory, chunk))
 
-        self._current_tx = Transaction(self._filenames[0])
-
-    def __next__(self) -> Transaction:
-        try:
-            return next(self._current_tx)
-        except StopIteration:
-            self._fileindex += 1
-            if len(self._filenames) > self._fileindex:
-                self._current_tx = Transaction(self._filenames[self._fileindex])
-                return next(self._current_tx)
-            else:
-                raise
+    def __next__(self) -> Ledgerchunk:
+        self._fileindex += 1
+        if len(self._filenames) > self._fileindex:
+            self._current_chunk = Ledgerchunk(self._filenames[self._fileindex])
+            return self._current_chunk
+        else:
+            raise StopIteration
```

Comment:
```suggestion
            raise
```

Reply:
@achamayou, got a runtime error when I removed the StopIteration "RuntimeError: No active exception to reraise".

Reply:
My bad, you're absolutely right, I was looking at the left of the diff and failed to spot this isn't an except path anymore. Sorry.